tests/virtual_machines_test.go

```diff
@@ -30,9 +32,34 @@ var _ = Describe("Virtual Machines", func() {
 			Body([]byte(fmt.Sprintf(linuxBridgeConfCRD, "linux-bridge", TestNamespace))).
 			Do()
 		Expect(result.Error()).NotTo(HaveOccurred())
+
+		By("Verify that there are no VMs in the cluster before starting test suite")
+		beforeAllVMList := &kubevirtv1.VirtualMachineList{}
+		err := testClient.VirtClient.List(context.TODO(), beforeAllVMList, &client.ListOptions{})
+		Expect(err).ToNot(HaveOccurred(), "Should successfully list add VMs")
+		Expect(len(beforeAllVMList.Items)).To(BeZero(), "There should be no VM's in the cluster before running test suite")
 	})
 
 	Context("Check the client", func() {
+		BeforeEach(func() {
+			By("Verify that there are no VMs left from previous tests")
+			currentVMList := &kubevirtv1.VirtualMachineList{}
+			err := testClient.VirtClient.List(context.TODO(), currentVMList, &client.ListOptions{})
+			Expect(err).ToNot(HaveOccurred(), "Should successfully list add VMs")
+			Expect(len(currentVMList.Items)).To(BeZero(), "There should be no VM's in the cluster before a test")
+
+			By("Clear vmWaitConfigMap configMap data before each test")
```

Comment:
You are not clearing then now, just getting them to clear later. Let's keep the `By` telling truth for easier debugging